codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -844,57 +843,58 @@ public void bind(SocketAddress localAddress, ChannelPromise promise) {
+                            public void operationComplete(ChannelFuture future) throws Exception {
+                                promise.setSuccess();
+                            }
+                        });
+                    }
                     return;
                 }
-                if (pendingClosePromise != null) {
-                    pendingClosePromise.addListener(new DelegatingChannelPromiseNotifier(promise));
-                    return;
+                closeInitiated = true;
+
+                closePending = false;
+                fireChannelReadPending = false;
+
+                // Only ever send a reset frame if the connection is still alive as otherwise it makes no sense at
+                // all anyway.
+                if (parent().isActive() && !streamClosedWithoutError && isStreamIdValid(stream().id())) {
+                    Http2StreamFrame resetFrame = new DefaultHttp2ResetFrame(Http2Error.CANCEL).stream(stream());
+                    write(resetFrame, unsafe().voidPromise());
+                    flush();
                 }
-                pendingClosePromise = promise;
-                try {
-                    closePending = false;
-                    fireChannelReadPending = false;
-
-                    // Only ever send a reset frame if the connection is still alive as otherwise it makes no sense at
-                    // all anyway.
-                    if (parent().isActive() && !streamClosedWithoutError && isStreamIdValid(stream().id())) {
-                        Http2StreamFrame resetFrame = new DefaultHttp2ResetFrame(Http2Error.CANCEL).stream(stream());
-                        write(resetFrame, unsafe().voidPromise());
-                        flush();
-                    }
 
-                    if (inboundBuffer != null) {
-                        for (;;) {
-                            Object msg = inboundBuffer.poll();
-                            if (msg == null) {
-                                break;
-                            }
-                            ReferenceCountUtil.release(msg);
+                if (inboundBuffer != null) {
+                    for (;;) {
+                        Object msg = inboundBuffer.poll();
+                        if (msg == null) {
+                            break;
                         }
+                        ReferenceCountUtil.release(msg);
                     }
+                }
 
-                    pipeline().fireChannelInactive();
-                    if (isRegistered()) {
-                        deregister(unsafe().voidPromise());
-                    }
-                    promise.setSuccess();
-                    closePromise.setSuccess();
-                } finally {
-                    pendingClosePromise = null;
+                pipeline().fireChannelInactive();
+                if (isRegistered()) {
+                    deregister(unsafe().voidPromise());
                 }
+                promise.setSuccess();
```

Comment:
the order of this line and the next line should be reversed to be consistent with `AbstractChannel`. see https://github.com/netty/netty/pull/7643